pkg/serverless/protocol.go

```diff
@@ -170,9 +178,12 @@ func (d *Daemon) Stop() {
 	}
 	d.stopped = true
 
-	// Wait for any remaining logs to arrive via the logs API before shutting down the HTTP server
-	log.Debug("Waiting to shut down HTTP server")
-	time.Sleep(shutdownDelay)
+	if !isTimeout {
+		// Wait for any remaining logs to arrive via the logs API before shutting down the HTTP server
+		log.Debug("Waiting to shut down HTTP server")
+		time.Sleep(shutdownDelay)
```

Comment:
This gives us 1 second to flush all data. In far regions, that might not be enough. How did we fine tune this?

Reply:
Is there a variable we can check here? Maybe we can busy poll, or use a channel?

Reply:
The 1s delay was already there. When we subscribe to the log, we specify that we want the logs payload sent to us at least every 1000ms using the `"timeoutMs": 1000`, so we're safe with the receiving part. For the processing part, we have already a wait group on it (readyWg) so I think we're 100% safe with that ?

Reply:
(and if it takes 2s or more, AWS will still shutdown the extension and logs will be lost forever)